### DISCUSSION TITLE
Pychron Data Collection Faults

### DISCUSSION THREAD

**Discussion by jirhiker** [NMGRL/pychron#2189] [General]
This discussion is for cataloging the various faults users experience during automated data collection so that a future version of pychron can better handle these faults

**Comment by jirhiker** [comment]
Most common problem is network connectivity issues (?)i.e. lost of connection via TCP. Please describe network issues and provide logs capturing the issues.

**Comment by cmmercer** [comment]
I've noticed one where a run will save (successfully, so network connectivity seems okay), but the queue hangs right after the save flag is cleared. Here are some relevant logs for a recent occurrence. See logs around these lines: 
```
DVCPersister                            : 2023-01-09 12:03:21,006 DEBUG     (Thread-103) clear save flag
pyExperiment                            : 2023-01-09 14:38:41,605 DEBUG     (MainThread) &&&& len __gloggers__ = 127
```
[pychron.current.log](https://github.com/NMGRL/pychron/files/10377142/pychron.current.log)
[a-01-A-811-001.log](https://github.com/NMGRL/pychron/files/10377143/a-01-A-811-001.log)

**Comment by cmmercer** [comment]
Had another hangup when executing a queue, during persistence of a run. Looks associated with a network connectivity issue while syncing/merging Git repos. Log attached.
[3103-46-001.log](https://github.com/NMGRL/pychron/files/10403698/3103-46-001.log)

**Comment by cmmercer** [comment]
Here is another error, just started occurring around the time a run is saved. In particular, there seems to be an sqlalchemy exception (see attached log, search for `'loaded_positions' `), but there are also some other errors it seems near the end of the log. This was an air shot, so many of them make sense (no K/Ca ratio), so maybe they're just informational. Interestingly, this run was persisted to the DB/GitHub just fine, but the run queue/Pychron crashed and closed after this run.

[a-01-A-966-001.log](https://github.com/NMGRL/pychron/files/10710637/a-01-A-966-001.log)
